crates/l3/appchain-core-contract-client/src/interfaces/operator.rs

```
@@ -0,0 +1,115 @@
+use common::LocalWalletSignerMiddleware;
+use common::{call_contract, invoke_contract};
+use starknet_accounts::Execution;
+use starknet_core::types::{FieldElement, StarknetError};
+use std::sync::Arc;
+
+pub struct Operator {
+    client: Arc<LocalWalletSignerMiddleware>,
+    address: FieldElement,
+}
+
+impl Operator {
+    pub fn new(address: FieldElement, client: Arc<LocalWalletSignerMiddleware>) -> Self {
+        Self { client, address }
+    }
+
+    pub async fn register_operator(
+        &self,
+        new_operator: FieldElement,
+    ) -> Result<Option<Execution<LocalWalletSignerMiddleware>>, StarknetError> {
+        let execution = invoke_contract(
+            &self.client,
+            self.address,
+            "register_operator",
+            vec![new_operator.into()],
+        )
+        .await;
+        Ok(Some(execution))
+    }
+
+    pub async fn unregister_operator(
+        &self,
+        new_operator: FieldElement,
+    ) -> Result<Option<Execution<LocalWalletSignerMiddleware>>, StarknetError> {
+        let execution = invoke_contract(
+            &self.client,
+            self.address,
+            "unregister_operator",
+            vec![new_operator.into()],
+        )
+        .await;
+        Ok(Some(execution))
+    }
+
+    pub async fn is_operator(&self) -> Result<bool, StarknetError> {
+        let result = call_contract(self.client.clone(), self.address, "is_operator").await;
+        match result {
+            Some(values) => {
+                if let Some(value) = values.first() {
+                    Ok(value.to_string() != String::from("0"))
+                } else {
+                    Err(StarknetError::ContractError)
+                }
+            }
+            None => Err(StarknetError::ContractError),
```

we should return a Result from call_contract instead of an Option and throw the exact error here?

---

I think we are returning error from call contract but in case there is any malformed request or any other issue, ContractError is raised, should we ignore this error and relying on error from call_contract would be sufficient?